### PR TITLE
Ngao4 daweix3/anti jam

### PR DESCRIPTION
**Pull request by ngao4**
Successfully test the anti jam. 
Fix the antijam code in the lib. 
Give a new example called m_2006_anti_jam.

### CODE REVIEW

**Review by jerrywrx — APPROVED** [review]

File: examples/motor/m2006_antijam.cc [hunk 1]
```diff
@@ -0,0 +1,101 @@
+/****************************************************************************
+ *                                                                          *
+ *  Copyright (C) 2022 RoboMaster.                                          *
```
Comment (AzulRadio) [inline examples/motor/m2006_antijam.cc:3]:
```suggestion
 *  Copyright (C) 2023 RoboMaster.                                          *
```

File: shared/libraries/motor.cc [hunk 2]
```diff
@@ -320,20 +325,22 @@ void ServoMotor::CalcOutput() {
...
-  int16_t command;
+  int command;
   float target_diff = (target_angle_ - servo_angle_ - cumulated_angle_) * transmission_ratio_;
   // v = sqrt(2 * a * d)
   uint32_t current_time = GetHighresTickMicroSec();
   if (!hold_) {
     float speed_max_start =
         (current_time - start_time_) / 10e6 * max_acceleration_ * transmission_ratio_;
     float speed_max_target = sqrt(2 * max_acceleration_ * abs(target_diff));
+    // print("%f\n", speed_max_target);
```
Comment (AzulRadio) [inline shared/libraries/motor.cc:336]:
```suggestion
```

File: shared/libraries/motor.cc [hunk 3]
```diff
@@ -320,20 +325,22 @@ void ServoMotor::CalcOutput() {
...
     float current_speed = speed_max_start > speed_max_target ? speed_max_target : speed_max_start;
     current_speed = clip<float>(current_speed, 0, max_speed_);
     command = omega_pid_.ComputeConstrainedOutput(
         motor_->GetOmegaDelta(sign<float>(target_diff, 0) * current_speed));
   } else {
     command = omega_pid_.ComputeConstrainedOutput(motor_->GetOmegaDelta(target_diff * 50));
+    // print("3");
```
Comment (AzulRadio) [inline shared/libraries/motor.cc:343]:
```suggestion
```

**Review by AzulRadio — CHANGES_REQUESTED** [review]
This example is simple enough to not have a description.

**Review by AzulRadio — APPROVED** [review]
LGTM

**Pull request merged**